libnetwork/network.go

```diff
@@ -2147,38 +2135,18 @@ func (n *Network) createLoadBalancerSandbox() (retErr error) {
 	return sb.EnableService()
 }
 
-func (n *Network) deleteLoadBalancerSandbox() error {
-	n.mu.Lock()
-	c := n.ctrlr
-	name := n.name
-	n.mu.Unlock()
-
-	sandboxName := n.lbSandboxName()
-	endpointName := n.lbEndpointName()
-
-	endpoint, err := n.EndpointByName(endpointName)
-	if err != nil {
-		log.G(context.TODO()).Warnf("Failed to find load balancer endpoint %s on network %s: %v", endpointName, name, err)
-	} else {
-		info := endpoint.Info()
-		if info != nil {
-			sb := info.Sandbox()
-			if sb != nil {
-				if err := sb.DisableService(); err != nil {
-					log.G(context.TODO()).Warnf("Failed to disable service on sandbox %s: %v", sandboxName, err)
-					// Ignore error and attempt to delete the load balancer endpoint
-				}
+// deleteLoadBalancer deletes the load-balancer sandbox attached to n.
+func (n *Network) deleteLoadBalancer(sb *Sandbox, endpoint *Endpoint) error {
+	info := endpoint.Info()
+	if info != nil {
+		sb := info.Sandbox()
+		if sb != nil {
+			if err := sb.DisableService(); err != nil {
+				log.G(context.TODO()).Warnf("Failed to disable service on sandbox %s: %v", sb.id, err)
+				// Ignore error and attempt to delete the load balancer endpoint
 			}
 		}
-
-		if err := endpoint.Delete(true); err != nil {
-			log.G(context.TODO()).Warnf("Failed to delete endpoint %s (%s) in %s: %v", endpoint.Name(), endpoint.ID(), sandboxName, err)
-			// Ignore error and attempt to delete the sandbox.
-		}
 	}
 
-	if err := c.SandboxDestroy(sandboxName); err != nil {
-		return fmt.Errorf("Failed to delete %s sandbox: %v", sandboxName, err)
-	}
-	return nil
+	return sb.delete(true)
```

Comment:
`c.SandboxDestroy()` is equivalent to `sb.delete(false)`. However, the original code was calling `endpoint.Delete(true)` before calling `c.SandboxDestroy()`. This delete operation is now done by `sb.delete(true)`.